cloudflare/resource_cloudflare_load_balancer_pool.go

```diff
@@ -253,15 +301,20 @@ func resourceCloudflareLoadBalancerPoolRead(d *schema.ResourceData, meta interfa
 	return nil
 }
 
-func flattenLoadBalancerOrigins(origins []cloudflare.LoadBalancerOrigin) *schema.Set {
+func flattenLoadBalancerOrigins(d *schema.ResourceData, origins []cloudflare.LoadBalancerOrigin) *schema.Set {
 	flattened := make([]interface{}, 0)
-	for _, o := range origins {
+	for idx, o := range origins {
 		cfg := map[string]interface{}{
 			"name":    o.Name,
 			"address": o.Address,
 			"enabled": o.Enabled,
 			"weight":  o.Weight,
 		}
+
+		if _, ok := d.GetOkExists(fmt.Sprintf("origins.%d.header", idx)); ok {
```

Comment:
thinking more about the test failure above, this isn't `GetOkExists` isn't going to work as the `idx` isn't ever going to find the correct hash. we may need to rethink this.

Reply:
we may just need to iterate instead using `(...).(*schema.Set).List()` however i'm not sure if that would still work 🤔 [`expandZoneLockdownConfig`](https://github.com/cloudflare/terraform-provider-cloudflare/blob/a117eb773c303ebbaf657000eb59d8e736809141/cloudflare/resource_cloudflare_zone_lockdown.go#L226) might be helpful